lib/plenario_web/controllers/api/detail_controller.ex

```diff
@@ -1,42 +1,64 @@
 defmodule PlenarioWeb.Api.DetailController do
   use PlenarioWeb, :api_controller
   import PlenarioWeb.Api.Utils, only: [render_page: 5, map_to_query: 2]
-  alias Plenario.Actions.MetaActions
   alias Plenario.{ModelRegistry, Repo}
+  alias Plenario.Actions.MetaActions
   alias PlenarioWeb.Controllers.Api.CaptureArgs
 
   defmodule CaptureColumnArgs do
     def init(opts), do: opts
 
     def call(conn, opts) do
-      columns = 
-        conn.params["slug"]
-        |> MetaActions.get()
-        |> MetaActions.get_column_names()
-      CaptureArgs.call(conn, opts ++ [fields: columns])
+      meta = MetaActions.get(conn.params["slug"], with_virtual_points: true) 
+      columns = MetaActions.get_column_names(meta)
+      vpfs = 
+        meta.virtual_points()
+        |> Enum.map(fn vpf -> vpf.name() end)
+      CaptureArgs.call(conn, opts ++ [fields: columns ++ vpfs])
+    end
+  end
```

Comment:
`CaptureArgs` throws out any query args with keys that aren't present in a dataset's column names. This bit of code instructs `CaptureArgs` to not throw out virtual point keys either.